### PR TITLE
Task-7: Filter blog using tags

### PR DESCRIPTION
**Pull request by mza-wtag**
[Task 7: Filter blog using tags](https://trello.com/c/EMuueRFx/11-task-7-filter-blog-using-tags)

**Small Description:** As a user I want to filter blogs by tags

## Types of changes
- [x] User can click on tags from the select menu to filter blogs.

## Review
Please have a close look at this PR and kindly put your valuable comments.

## Screen recording
[Screencast from 02-21-2024 12:19:38 PM.webm](https://github.com/mza-wtag/my-blog-react-app/assets/151821213/9a2404fe-debc-4c16-8d38-7e5e52d5cf67)

### CODE REVIEW

File: src/components/BlogList/BlogList.jsx [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 
 const BlogList = ({ blogs }) => {
-  const queryString = useSelector((state) => state.search.searchQuery);
-  const searchedBlogs = blogs?.filter((blog) =>
-    blog?.title?.toLowerCase().includes(queryString)
-  );
+  const searchQuery = useSelector((state) => state.search.searchQuery);
+  const filteredTag = useSelector((state) => state.filter.filteredTag);
+
+  const filteredBlogs = blogs.filter((blog) => {
```
Comment (nmi-wtag) [inline src/components/BlogList/BlogList.jsx:12]:
```suggestion
  const filteredBlogs = blogs?.filter((blog) => {
```

**Review by nmi-wtag — APPROVED** [review]

File: src/components/BlogList/BlogList.jsx (hunk 1 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx:12]: done

File: src/components/BlogList/BlogList.jsx [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+        filteredBlogs.map((blog) => <BlogCard key={blog.id} blog={blog} />)
       ) : (
         <NotFound text="No Blogs Found." />
       )}
     </div>
   );
 };
 
+BlogList.propTypes = {
+  blogs: PropTypes.array.isRequired,
```
Comment (maa-wtag) [inline src/components/BlogList/BlogList.jsx]: it should not be required in the initial stage as no Blog will be there

File: src/components/FilterBlogs/FilterBlogs.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const handleChange = (event) => {
+    const tag = event.target.value;
+    setCurrentTag(tag);
+    dispatch(filterByTag(tag));
+  };
+
+  return (
+    <div className="filter-blogs">
+      <label className="filter-blogs__label">Filter Blogs By Tags:</label>
+      <select
```
Comment (maa-wtag) [inline src/components/FilterBlogs/FilterBlogs.jsx]: please use the SelectBox component

**Review by maa-wtag — CHANGES_REQUESTED** [review]

File: src/components/BlogList/BlogList.jsx (hunk 2 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx]: done

File: src/components/FilterBlogs/FilterBlogs.jsx (hunk 3 above)
Comment (mza-wtag) [reply src/components/FilterBlogs/FilterBlogs.jsx]: done

File: src/components/FilterBlogs/FilterBlogs.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { filterByTags } from "@actions/blogActions";
+import SelectBox from "@components/SelectBox/SelectBox";
+import "./filterBlogs.scss";
+
+const FilterBlogs = ({ tags }) => {
+  const dispatch = useDispatch();
+  const selectedTags = useSelector((state) => state?.filter?.filteredTags);
+
+  const handleChangeTags = (selectedTags) => {
+    const tagsArray = selectedTags.map((tag) => tag.value);
```
Comment (maa-wtag) [inline src/components/FilterBlogs/FilterBlogs.jsx:13]:
```suggestion
    const tagsArray = selectedTags?.map((tag) => tag.value);
```

**Review by maa-wtag — APPROVED** [review]
Please make the filter component here as the same design like the Multi Select used in the Adding a new blog post

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**